### DISCUSSION TITLE
InputNumber has new undesired option modifyValueOnUpDownArrow enabled by default - Primefaces 14.0.0

### DISCUSSION THREAD

**Discussion by hangrybear666** [primefaces/community#2199] [PrimeFaces] [answered]
The `<p:inputNumber/>` Element now increments and decrements its values by 1 each time up or down arrow are pressed
This is new behavior compared to our prior Version Primefaces 10.0 and undesired, we now have to do a manual override via Primefaces API each time a user focuses an inputNumber field (because our customer uses arrow keys for navigation between inputs).

```
<p:inputNumber id="exampleId" 
    value="#{row.persId}"
    pt:rowIndex="#{rowIndex}"
    modifyValueOnWheel="false" 
    widgetVar="exampleId#{rowIndex}"
>
    <p:ajax event="focus" onstart="PF('exampleId#{rowIndex}').autonumeric.options.modifyValueOnUpDownArrow(false);" />
</p:inputNumber>
```

Ideally, you guys would expose this option in the `InputNumberBase.class` so it can be used on the `p:inputNumber` Element itself, similar to how `modifyValueOnWheel` behaves.

```
public abstract class InputNumberBase extends AbstractPrimeHtmlInputText implements Widget, InputHolder {
    public enum PropertyKeys {
        ...
        modifyValueOnWheel,
        modifyValueOnUpDownArrow, // add this flag
        ...
    }
}
```

**Comment by melloware** [comment]
@hangrybear666 i don't think that is correct?  For keyboard accessibility for visually impaired the arrows keys should increment up and down.

And I don't see any `modifyValueOnUpDownArrow` here at AutoNumeric: https://autonumeric.org/guide  only `modifyOnWheel`

**Reply by melloware** [reply]
Actually found it: https://github.com/autoNumeric/autoNumeric/issues/719

`modifyValueOnUpDownArrow ` he needs to update his docs!

**Reply by hangrybear666** [reply]
> @hangrybear666 i don't think that is correct? For keyboard accessibility for visually impaired the arrows keys should increment up and down.
> 
> And I don't see any `modifyValueOnUpDownArrow` here at AutoNumeric: https://autonumeric.org/guide

Perhaps it is desired for accessibility but it is a new feature compared to Primefaces 10. And broke some of our existing code.
If you check within `/META-INF/resources/primefaces/inputnumber/inputnumber.js` of your `primefaces-14.0.0-jakarta.jar`
you will find the following lines:
```
  modifyValueOnUpDownArrow: function (g) {
      a.settings.modifyValueOnUpDownArrow = g;
      return a;
  },
```
```
c["default"].isBoolean(b.modifyValueOnUpDownArrow) ||
c["default"].throwError(
    "The increment/decrement on up and down arrow keys 'modifyValueOnUpDownArrow' is invalid ; it should be either 'true' or 'false', [".concat(b.modifyValueOnUpDownArrow, "] given.")
);
```

```
if (this.settings.modifyValueOnUpDownArrow && (this.eventKey === f["default"].keyName.UpArrow || this.eventKey === f["default"].keyName.DownArrow)) {
    this.upDownArrowAction(a);
    return;
}
```

```
key: "upDownArrowAction",
value: function (a) {
	if (!(this.formulaMode || this.settings.readOnly || this.domElement.readOnly || this.domElement.disabled)) {
		var b = !1,
			e = !1;
		this.eventKey === f["default"].keyName.UpArrow
			? (b = !0)
			: this.eventKey === f["default"].keyName.DownArrow
			? (e = !0)
			: c["default"].throwError("Something has gone wrong since neither an Up or Down arrow key is detected, but the function was still called!");
		this._wheelAndUpDownActions(a, b, e, this.settings.upDownStep);
	}
},
```
[unminified-inputnumber.zip](https://github.com/user-attachments/files/15973768/unminified-inputnumber.zip)

**Reply by melloware** [reply, marked as answer]
Issue created: https://github.com/primefaces/primefaces/issues/12187

PR submitted: https://github.com/primefaces/primefaces/pull/12188

**Reply by melloware** [reply]
you can MonkeyPatch this yourself with

```js
if (PrimeFaces.widget.InputNumber) {
    PrimeFaces.widget.InputNumber= PrimeFaces.widget.InputNumber.extend({
        init: function(cfg) {
            cfg.modifyValueOnUpDownArrow = false;
            this._super(cfg);
        }
    });
};
```
